onnxruntime/python/tools/transformers/optimizer.py

```diff
@@ -304,7 +304,7 @@ def optimize_model(input,
     if disable_onnxruntime is False:
         if opt_level > 1:  # Optimization specified for an execution provider.
             temp_model_path = optimize_by_onnxruntime(input, use_gpu=use_gpu, opt_level=opt_level)
-        elif run_onnxruntime:
+        elif opt_level == 1 and run_onnxruntime:
```

Comment:
I think we need to consolidate 3 settings: opt_level, run_onnxruntime, disable_onnxruntime.

Considering other cases like opt_level==1 but run_onnxruntime is False. There is still confusion there.

I would suggest to not change it in the PR to avoid confusing. Such change need more tests, and can be in another PR.

Reply:
I see, will back this out and go over the change for a different PR.